analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/LoopDetector.cs

```diff
@@ -83,13 +83,22 @@ void MergeWithIntersectingLoops(int[] path, int last)
         }
     }
 
-    private static IEnumerable<int> Successors(ControlFlowGraph cfg, int block)
+    private static IEnumerable<int> Successors(ControlFlowGraph cfg, int index)
     {
-        var successors = cfg.Blocks[block].SuccessorBlocks.Select(x => x.Ordinal);
-        if (cfg.Blocks[block].EnclosingRegion(ControlFlowRegionKind.Try) is { } tryRegion)
+        var block = cfg.Blocks[index];
+        var successors = block.SuccessorBlocks.Select(x => x.Ordinal);
+        if (block.EnclosingRegion(ControlFlowRegionKind.Try) is { } tryRegion)
         {
             successors = successors.Concat(tryRegion.ReachableHandlers().Select(x => x.FirstBlockOrdinal));
         }
+        if (block.EnclosingRegion(ControlFlowRegionKind.Finally) is { } finallyRegion)
+        {
+            var tryFinally = finallyRegion.EnclosingRegion;
+            successors = successors.Concat(
+                cfg.Blocks.Where(x => x.Ordinal >= tryFinally.FirstBlockOrdinal && x.Ordinal <= tryFinally.LastBlockOrdinal)
+                    .SelectMany(x => x.SuccessorBlocks.Select(x => x.Ordinal))
+                    .Where(x => !cfg.Blocks[x].EnclosingRegions().Contains(tryFinally)));
```

Comment:
This will be way faster. 
```suggestion
                    .Where(x => x < tryFinally.FirstBlockOrdinal || x > tryFinally.LastBlockOrdinal)
```